lib/tofuenv-version-name.sh

```diff
@@ -10,10 +10,14 @@ function tofuenv-version-name() {
       && log 'debug' "TOFUENV_VERSION_FILE retrieved from tofuenv-version-file: ${TOFUENV_VERSION_FILE}" \
       || log 'error' 'Failed to retrieve TOFUENV_VERSION_FILE from tofuenv-version-file';
 
-    TOFUENV_VERSION="$(cat "${TOFUENV_VERSION_FILE}" || true)" \
-      && log 'debug' "TOFUENV_VERSION specified in TOFUENV_VERSION_FILE: ${TOFUENV_VERSION}";
-
-    TOFUENV_VERSION_SOURCE="${TOFUENV_VERSION_FILE}";
+    if [[ -f "${TOFUENV_VERSION_FILE}" ]]; then
+        TOFUENV_VERSION="$(cat "${TOFUENV_VERSION_FILE}" || true)" \
+            && log 'debug' "TOFUENV_VERSION specified in TOFUENV_VERSION_FILE: ${TOFUENV_VERSION}";
+        TOFUENV_VERSION_SOURCE="${TOFUENV_VERSION_FILE}";
+    else
+        TOFUENV_VERSION="${TOFUENV_TOFU_DEFAULT_VERSION:-"latest"}"
```

Comment:
The default value should be empty if no version is set in `TOFUENV_TOFU_DEFAULT_VERSION` variable. Otherwise, it will always trigger the installation of the latest version.

Reply:
Isn't this what the documentation for `tofuenv install` says should happen, though?

> If no parameter is passed, the version to use is resolved automatically via [TOFUENV_TOFU_VERSION environment variable](https://github.com/tofuutils/tofuenv#TOFUENV_TOFU_VERSION) or [.opentofu-version files](https://github.com/tofuutils/tofuenv#opentofu-version-file), in that order of precedence, i.e. TOFUENV_TOFU_VERSION, then .opentofu-version. **The default is latest if none are found**.

Reply:
Yes, but `tofuenv install` already works this way. 
It calls `tofuenv-resolve-version` first - https://github.com/tofuutils/tofuenv/blob/main/libexec/tofuenv-install#L67C13-L67C36, and then installs `latest` if no other version has been specified - https://github.com/tofuutils/tofuenv/blob/main/libexec/tofuenv-resolve-version#L97

Actually, `tofuenv install` command doesn't use `tofuenv version-name` script at all (tofuenv version-name command just checks and returns opentofu version currently used in a current directory). So to add the support of a new `TOFUENV_TOFU_DEFAULT_VERSION` variable during installation process, you'll need to add a correspoding change to lib/tofuenv-install.sh script as well.

Reply:
Just noticed that `tofuenv version-name` command is not described in the documentation, we'll add it there. But the output of `tofuenv help` returns the following description:
>  version-name  Print current version of OpenTofu
So this command should not trigger any installation and should just show the current OpenTofu version. If it's set to `latest` by default (when no version is specified in `.opentofu-version` file, in `version` file, or in `TOFUENV_TOFU_DEFAULT_VERSION` variable), it's illogical, when a user doesn't have any OpenTofu installed. Moreover, setting it to `latest` by default if no version is specified anywhere will trigger OpenTofu installation, when a user just wants to check if they have any versions installed, but not to install anything.
So if no OpenTofu version is specified in the list of sources to check, it should remain empty when calling `tofuenv version-name`.

However, you're right that `tofuenv install` command should install `latest` version if no version is specified - but it's set in a different script, to which I attached the link above. That would be great if you could add support for the `TOFUENV_TOFU_DEFAULT_VERSION` there 🙂

Reply:
Thank you for the explanation.  After looking over the files you mentioned I see my mistake.

At the same time, I wonder if perhaps the behaviour of this change is correct?  It depends on what one expects `tofuenv install` to do.  In my case, I expected it to install whatever version the current directory specified. But if the actual functionality is to install the specified version, or latest, then it is correct (although my README update is wrong).

As it is now, the current behavior does what I want in that `tofu ...` will install the version specified in the directory or fallback to an environment variable. However, as you mention, it does not for `tofu install`.

What would you like the behaviour of these to operations to be?

Reply:
Currently, as you mentioned, `tofuenv install` command first checks if any opentofu version is set in `TOFUENV_TOFU_VERSION` environment variable, and then in `.opentofu-version` file, and if no versions are found - use `latest`.
But as you mentioned in README, it would be great to support default version setting. In this case, `tofuenv install` command without any version passed will check the following sources:

> - [TOFUENV_TOFU_VERSION environment variable](#TOFUENV_TOFU_VERSION)
> - [.opentofu-version files](#opentofu-version-file)
> - [TOFUENV_TOFU_DEFAULT_VERSION](#TOFUENV_TOFU_DEFAULT_VERSION)
> - The default of `latest`.

So we will still have `latest` version if no version was specified, but also there will be an ability to set a default one using `TOFUENV_TOFU_DEFAULT_VERSION` environment variable in, for ex., some CI/CD pipeline

Reply:
Great! Just confirming the desired behaviour.  I'll finish this change it ASAP